### DISCUSSION TITLE
Adding constraints to parameters alone

### DISCUSSION THREAD

**Discussion by jbuzhar** [UW-ACL/SCPToolbox.jl#12] [Q&A]
Hello, 

I am new to the toolbox and was trying to implement it on my own problem after looking through the examples.  I would like to add a convex constraint to a parameter, p.  This constraint is not a function of the states or controls -- only the parameter.  A simple example might be that I want to provide an upper or lower bound on the final time in a free final time formulation.  Is there a nice way to implement such a constraint using the toolbox?  

At first I was thinking that I could use `problem_set_X!` or `problem_set_U!` to add these, but these appear to add N constraints to the problem, one for each timestep, which is unnecessary in my case since the parameter constraint should only be added once.  Is there a better way to do this?  

Thanks, 

Jake

**Comment by sandeepkumar575** [comment]
Hey!
I know this was a long time ago but I've just recently started looking into the toolbox. 
Maybe you can add it as a non-convex path constraint ( s(t, x(t), u(t), p) as mentioned in the tutorial paper ) which is a function of all variables and parameters? Did you figure out how to do it in a different way?
